src/main/java/bio/terra/service/dataset/flight/ingest/SkippableStep.java

```diff
@@ -0,0 +1,49 @@
+package bio.terra.service.dataset.flight.ingest;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import java.util.function.Predicate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class SkippableStep implements Step {
+
+  private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+  private final Predicate<FlightContext> skipCondition;
+
+  public SkippableStep() {
+    this(x -> false);
+  }
+
+  public SkippableStep(Predicate<FlightContext> skipCondition) {
+    this.skipCondition = skipCondition;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext)
+      throws InterruptedException, RetryException {
+    if (skipCondition.test(flightContext)) {
+      logger.info("Skipping {} step based on flight context state", this.getClass().getName());
```

Comment:
This is really cool!

Reply:
Thanks! @snf2ye and I worked hard on it!